### DISCUSSION TITLE
a optional text field for TouchScreenButton

### DISCUSSION THREAD

**Discussion by shahriarlabib000** [godotengine/godot-proposals#9971] [GUI]
When trying to make a prototype for a game we often don't have all the texture for a button. so I typically set the texture to icon.svg. But if I have multiple buttons 
it becomes hard to differentiate. If there was a easier way to add text to a `TouchScreenButton` it'll make it easier to prototype. It'll probably also help while making menus or settings ui for a game. I could use `Button` Node but I'll not get the multi touch functionality. I could also just add a `label` on top of the `TouchScreenButton` but it'll become tedious really quickly.

**Comment by AThousandShips** [comment]
See also:
* https://github.com/godotengine/godot-proposals/discussions/9405

**Reply by shahriarlabib000** [reply]
i think using `label` is kinda tedious  if you have multiple buttons

**Reply by AThousandShips** [reply]
I didn't say anything about the suggestions there, just coordinating with a related proposal

**Reply by shahriarlabib000** [reply]
ooh,sorry

**Comment by theraot** [comment]
You say it is hard to differentiate because they all have the same texture... For that, using `modulate` is what comes to mind.

However, I'm going to suggest you pick some CC0 texture pack and use that. I'm thinking of these: https://www.kenney.nl/assets/tag:interface

**Reply by shahriarlabib000** [reply]
but wouldn't it be even better to have a text field?

**Reply by theraot** [reply]
@shahriarlabib000 I suppose. I was thinking on what can you do today. On the flip side... Wouldn't it be even better to have multitouch in all `Control`s?

**Reply by Calinou** [reply]
- See https://github.com/godotengine/godot-proposals/issues/3976.

**Reply by shahriarlabib000** [reply]
@Calinou has this been implemented officially or is it being considered being implemented? also does this mean the `Button` node will support multi touch?

**Reply by AThousandShips** [reply]
It's a proposal, please see there for more details, the answers to your questions should be there
